js/src/annotations/osd-svg-overlay.js

```diff
@@ -303,60 +307,80 @@
-            }
-            // if item is part of shape it is editable
-            // part of shape items only appear when the shape is selected
-            if(hitResult.item._name.toString().indexOf(this.overlay.partOfPrefix) !== -1){
-              overlayEditable = true;
-            }
-            if (!overlayEditable) {
-              return;
-            }
+        //}
+
+        if(this.overlay.mode !== 'create'){
+          this.overlay.mode = '';
+          this.currentTool = null;
+        }
+
+        if (hitResult && this.overlay.mode !== 'create') {
+
+          var overlayEditable = false;
+          if (typeof hitResult.item.data.editable !== 'undefined') {
+            overlayEditable = hitResult.item.data.editable; // TODO should remove this editable variable when persisting the svg (will reduce length of string/memory)
           }
+          // if item is part of shape it is editable
+          // part of shape items only appear when the shape is selected
+          if (hitResult.item._name.toString().indexOf(this.overlay.partOfPrefix) !== -1) {
+            overlayEditable = true;
+          }
+          if (!overlayEditable) {
+            return;
+          }
+
         }
+
         if (hitResult && (!this.overlay.currentTool || (hitResult.item._name.toString().indexOf(this.overlay.currentTool.idPrefix) === -1 && this.overlay.mode === ''))) {
           var prefix = hitResult.item._name.toString();
           prefix = prefix.substring(0, prefix.indexOf('_') + 1);
+
+          // nasty workaround some names contain `_` inside their name
+          var longPrefix = hitResult.item._name.toString().split('_');
+
+          longPrefix = longPrefix[0] + '_' + longPrefix[1] + '_';
+
           for (var j = 0; j < this.overlay.tools.length; j++) {
-            if (this.overlay.tools[j].idPrefix === prefix) {
+
+            if (this.overlay.tools[j].idPrefix === prefix || this.overlay.tools[j].idPrefix === longPrefix) {
               this.overlay.eventEmitter.publish('toggleDrawingTool.' + this.overlay.windowId, this.overlay.tools[j].logoClass);
               break;
             }
           }
         }
+
         if (this.overlay.currentTool) {
           this.overlay.currentTool.onMouseDown(event, this.overlay);
-          if (this.overlay.mode === 'translate' || this.overlay.mode === 'deform' || this.overlay.mode === 'edit') {
-            if (this.overlay.path && this.overlay.path.data.annotation) {
-              var inArray = false;
-              for (var i = 0; i < this.overlay.editedPaths.length; i++) {
-                if (this.overlay.editedPaths[i].name === this.overlay.path.name) {
-                  inArray = true;
-                  break;
-                }
-              }
-              if (!inArray) {
-                this.overlay.editedPaths.push(this.overlay.path);
-              }
-            }
-          }
+          // should check if this is used anywhere and remove it if not used
```

Comment:
I don't think so.  I didn't write that - I think that was already there when I started adding the new edit code, so if it doesn't appear to be used anywhere, I think it can go away.